component/componenterror/errors.go

```diff
@@ -36,14 +36,18 @@ var (
 // CombineErrors converts a list of errors into one error.
 func CombineErrors(errs []error) error {
 	numErrors := len(errs)
+	if numErrors == 0 {
```

Comment:
Nice cleanup but it's out of scope of this PR :)

Reply:
It was not necessary out of scope, because I found that a separate check was in place for num errors = 0 in the jaeger code, mostly because this was not readable. Will create a separate PR, but gave context.